packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -141,9 +141,10 @@ export class TeamsBotImpl {
         CheckThrowSomethingMissing(ConfigNames.RESOURCE_GROUP, this.config.provision.resourceGroup);
         CheckThrowSomethingMissing(ConfigNames.LOCATION, this.config.provision.location);
         CheckThrowSomethingMissing(ConfigNames.SKU_NAME, this.config.provision.skuName);
-        
+        CheckThrowSomethingMissing(CommonStrings.SHORT_APP_NAME, this.ctx?.app.name.short);
+
         if (!this.config.provision.siteName) {
-            this.config.provision.siteName = ResourceNameFactory.createCommonName(this.ctx?.app.name.short);
+            this.config.provision.siteName = ResourceNameFactory.createCommonName(this.ctx?.app.name.short!, this.config.resourceNameSuffix, MaxLengths.WEB_APP_SITE_NAME);
```

Comment:
I would suggest not use ! as possible since it is error-prone

Reply:
Agree, let me update with the following comment.